packages/manager-react-components/src/components/filters/filter-add.component.tsx

```diff
@@ -30,6 +36,7 @@ export function FilterAdd({ columns, onAddFilter }: Readonly<FilterAddProps>) {
     columns?.[0]?.comparators?.[0] || FilterComparator.IsEqual,
   );
   const [value, setValue] = useState('');
+  const [dateValue, setDateValue] = useState<Date>(null);
```

Comment:
I think we should have only one value and other should have computed values or else we will duplicate code only for different types

Or we should think of another way to handle multiple types

Reply:
I tried using only one value but it was make the code harder to write and to read but several parsing/formatting of the date.
So I figured out it was better to separate the state by type in this case.
2 types is still easy to maintain in such a component.
But if one day more types come up, we should, indeed, think about refactoring it

Reply:
The value should be a Date, I set null to keep it consistent with the expected type